TaskHouseApi/Persistence/Repositories/Repository.cs

```diff
@@ -66,33 +66,45 @@ public void UpdatePart(T baseModel, string[] nameOfPropertysToIgnore)
             bool IsModified = false;
             foreach (PropertyInfo property in propertyInfos)
             {
-                /// If the as ICollection or name is Id do nothing
-                if (property.GetValue(baseModel) is ICollection || property.Name == "Id")
+
+                /// If the name is Id or the propertyname should be ignored or the type of a Reference do nothing 
+                if (property.Name == "Id" || nameOfPropertysToIgnore.Contains(property.Name) || property.PropertyType == typeof(Reference))
                 {
                     continue;
                 }
 
+                /// Gets the property value
+                var propertyValue = property.GetValue(baseModel);
+
                 /// Attach basemodel
                 if (!IsModified)
                 {
                     dbSet.Attach(baseModel);
                     IsModified = true;
                 }
 
-                /// Gets the property value
-                var propertyValue = property.GetValue(baseModel);
-
                 /// If property name is in ignore
                 /// or is null
                 /// don't change the value
                 if (nameOfPropertysToIgnore.Contains(property.Name) || propertyValue == null)
                 {
-                    context.Entry(baseModel).Property(property.Name).IsModified = false;
                     continue;
                 }
 
-                /// Marks the property as modified
-                context.Entry(baseModel).Property(property.Name).IsModified = true;
+                /// If property is a ICollection
+                /// don't change the value
+                if (property.PropertyType == typeof(ICollection))
+                {
+                    context.Entry(baseModel).Collection(property.Name).IsModified = false;
+                }
+
+                /// If property is is a string
+                /// change the value
+                if (property.PropertyType == typeof(string) || property.PropertyType == typeof(double) || property.PropertyType == typeof(int))
```

Comment:
```suggestion
                if (property.PropertyType == typeof(string) || property.PropertyType.IsValueType)
```
To cover more scenarios.